src/services/editor.service.ts

```diff
@@ -5,9 +5,17 @@ import { FormatService } from './format.service';
 import { SpecificationService } from './specification.service';
 
 import state from '../state';
+import { SocketClient } from './socket-client.service';
 
 export type AllowedLanguages = 'json' | 'yaml' | 'yml';
 
+export interface UpdateState {
```

Comment:
Note: If you agree with my suggestion but don't want to include anything on this PR, just tell me and I will open an issue instead.

I would like to suggest we add a new field where we specify the update mechanism to follow. For example, the current mechanism is replacing the whole content. However, we might want to do Patching eventually instead (using json diff, for example). This will reduce a lot the payload on the requests.
There is always the option to send a full update that will sync both sides, etc.

Reply:
Yeah go ahead and create a new issue. Too much for this one 😅